sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -481,8 +479,12 @@ private Mono<RxDocumentServiceResponse> invokeAsyncInternal(RxDocumentServiceReq
     }
 
     private Mono<RxDocumentServiceResponse> invokeAsync(RxDocumentServiceRequest request) {
-        Callable<Mono<RxDocumentServiceResponse>> funcDelegate = () -> invokeAsyncInternal(request).single();
-        return BackoffRetryUtility.executeRetry(funcDelegate, new WebExceptionRetryPolicy(BridgeInternal.getRetryContext(request.requestContext.cosmosDiagnostics)));
+
+        final WebExceptionRetryPolicy policyInstance = new WebExceptionRetryPolicy(BridgeInternal.getRetryContext(request.requestContext.cosmosDiagnostics));
+        return BackoffRetryUtility.executeRetry(() -> {
+            policyInstance.onBeforeSendRequest(request);
+            return invokeAsyncInternal(request).single();
```

Comment:
`invokeAsyncInternal()` already returns a `Mono`, we don't need to call `.single()` on it.

Reply:
Fixed it.